### PR TITLE
Introduced Nested Template Support in Prompts.

### PR DESCRIPTION
**Pull request by kladaFOX**
# Changelog for Version 1.1.1

**Release Date:** [11th Oct 2024]

**New Features:**

* **Nested Template Support in Prompts**
  * You can now organize your prompt files in nested directories and render them using the `Spectre::Prompt.render` method.
  * **Example**: To render a template from a nested folder:
    ```ruby
    Spectre::Prompt.render(template: 'nested/folder/nested', locals: { query: 'What is AI?' })
    ```
  * This feature allows for better organization and scalability when dealing with multiple prompt categories and complex scenarios.

### CODE REVIEW

File: lib/spectre/prompt.rb [hunk 1]
```diff
@@ -5,107 +5,122 @@
...
-      when Hash
-        value.transform_values { |v| preprocess_locals(v) } # Recurse into hash values
-      when Array
-        value.map { |item| preprocess_locals(item) } # Recurse into array items
-      else
-        value
+      # Detects the appropriate path for prompt templates
+      def detect_prompts_path
+        if Spectre.root
+          File.join(Spectre.root, 'app', 'spectre', 'prompts')
```
Comment (matthewblack) [inline lib/spectre/prompt.rb:54]: Is the only going use case for `Spectre.root` to be able to handle a custom prompts path?

If so, should we consider having the config not just the root, but a full relative path to the prompts directly?

**Review by matthewblack — COMMENTED** [review]
Awesome. One comment / question related to `Spectre.root`

File: lib/spectre/prompt.rb (hunk 1 above)
Comment (matthewblack) [reply lib/spectre/prompt.rb:54]: Cool!

**Review by matthewblack — APPROVED** [review]
👍

**Comment by kladaFOX** [comment]
@matthewblack take a look at new prompts feature `Nested Template Support in Prompts`
Now you can pass nested folders for templates as so: 
```ruby
Spectre::Prompt.render(template: 'nested/folder/nested', locals: { query: 'What is AI?' })
```

**Review by matthewblack — CHANGES_REQUESTED** [review]
Awesome!

For the example, let's update it to be this:

```
app/
  spectre/
    prompts/
      rag/
        system.yml.erb
        user.yml.erb
      classificiation/
        intent/
          system.yml.erb
          user.yml.erb
        entity/
          system.yml.erb
          user.yml.erb
```

**Pull request merged**